### PR TITLE
Small improvements and updates to `default.tensor` based on `quimb 1.8.2`

### PR DESCRIPTION
**Pull request by PietropaoloFrisoni**
**Context:** `quimb 1.8.2` has been released recently. This PR implements small improvements and updates to `default.tensor` based on new features introduced in the new version of `quimb`.

**Description of the Change:** As above.

**Benefits:** One fewer package to import.

**Possible Drawbacks:** None that I can think of.

**Related GitHub Issues:** None.

**Related Shortcut Stories:** [sc-65970]

### CODE REVIEW

File: pennylane/devices/default_tensor.py [hunk 1]
```diff
@@ -308,7 +307,7 @@ def circuit(theta, phi, num_qubits):
 
                 phi = 0.1
                 depth = 10
-                num_qubits = 25
+                num_qubits = 100
```
Comment (AmintorDusko) [inline pennylane/devices/default_tensor.py:310]: Ambitious! Nice! :smile:

File: pennylane/devices/default_tensor.py [hunk 2]
```diff
@@ -763,7 +762,9 @@ def _local_expectation(self, matrix, wires) -> float:
         """
 
         # We need to copy the quimb circuit since `local_expectation` modifies it.
-        qc = copy.deepcopy(self._quimb_circuit)
+        # If there is only one measurement and we don't wanto to keep track of the state
```
Comment (AmintorDusko) [inline pennylane/devices/default_tensor.py:765]:
```suggestion
        # If there is only one measurement and we don't want to keep track of the state
```

**Review by dwierichs — APPROVED** [review]
LGTM!

File: pennylane/devices/default_tensor.py [hunk 3]
```diff
@@ -501,7 +517,7 @@ def draw(self, color="auto", **kwargs):
 
         .. code-block:: python
 
-            dev = qml.device("default.tensor", method="tn")
+            dev = qml.device("default.tensor", method="tn", contract=False)
```
Comment (Qottmann) [inline pennylane/devices/default_tensor.py:520]: lol yeah good catch!

**Review by Qottmann — APPROVED** [review]
Looks good 👍

Do I see it right the changes are

- deepcopy -> quimb internal copy method
- draw examples set contract=False
- code example set 100 instead of 25 qubits
- typo `"True"` >> `True`

In case I didnt miss anything, looks good to me 👍

**Comment by PietropaoloFrisoni** [comment]
@Qottmann Yes, plus one more typo spotted by @AmintorDusko

**Review by AmintorDusko — APPROVED** [review]
Nice job!

**Pull request merged**